From 37f6a125e7d21f15dcb14b8efbed2787c0b791d8 Mon Sep 17 00:00:00 2001
From: Dylan Yip <dylan.yip@xilinx.com>
Date: Tue, 7 Jul 2020 10:55:17 -0700
Subject: [PATCH 1427/1851] v4l: xilinx: scd: Allow setting field on MM SCD
 subdev

commit 4fa0325be4f8cd25a9235afdf3e45c8c62f6cf84 from
https://github.com/Xilinx/linux-xlnx.git

Currently, SCD driver is ignoring the field in S_FMT. In memory based
SCD, the IP can take any field because it is agnostic to field. So, no
design change is needed and users can just set the field of the subdev.

For stream based SCD, there is no fid pin so currently we do not support
stream based interlaced SCD.

Signed-off-by: Dylan Yip <dylan.yip@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../media/platform/xilinx/xilinx-scenechange-channel.c    | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/media/platform/xilinx/xilinx-scenechange-channel.c b/drivers/media/platform/xilinx/xilinx-scenechange-channel.c
index a3c841a9f3aa..dba4fc3ea9d3 100644
--- a/drivers/media/platform/xilinx/xilinx-scenechange-channel.c
+++ b/drivers/media/platform/xilinx/xilinx-scenechange-channel.c
@@ -90,6 +90,14 @@ static int xscd_set_format(struct v4l2_subdev *subdev,
 	format->height = clamp_t(unsigned int, fmt->format.height,
 				 XSCD_MIN_HEIGHT, XSCD_MAX_HEIGHT);
 	format->code = fmt->format.code;
+	/*
+	 * If memory based, SCD can support interlaced alternate mode because
+	 * SCD is agnostic to field. If stream based, SCD does not have a fid
+	 * pin so interlaced alternate mode is not supported there.
+	 */
+	if (chan->xscd->memory_based)
+		format->field = fmt->format.field;
+
 	fmt->format = *format;
 
 	return 0;
-- 
2.31.1

